### PR TITLE
Remove deprecated APIs and modules

### PR DESCRIPTION
**Pull request by satya164**
We've deprecated these APIs for quite a few releases and we should be able to get rid of them now.

Remove following deprecated modules/components
 - AppStateIOS
 - ActivityIndicatorIOS
 - IntentAndroid
 - SliderIOS
 - SwitchAndroid
 - SwitchIOS
 - LinkingIOS

Update following modules to remove callback support
 - Clipboard
 - NetInfo

cc @bestander

### CODE REVIEW

**Comment by facebook-github-bot-2** [comment]
By analyzing the blame information on this pull request, we identified @davidaurelio and @mkonicek to be potential reviewers.

**Comment by ide** [comment]
This seems fine -- we need to mention this in the breaking changes of the release notes and mention what the replacement APIs are.

**Comment by facebook-github-bot-1** [comment]
@satya164 updated the pull request - [view changes](https://github.com/facebook/react-native/pull/9891/files/ea77f7057f82104cfabcdb020e963db13595e99a..6d227378b82fef681a5027bd2130d2dc2b20f4c5)

**Comment by facebook-github-bot-5** [comment]
@satya164 updated the pull request - [view changes](https://github.com/facebook/react-native/pull/9891/files/6d227378b82fef681a5027bd2130d2dc2b20f4c5..329509ebd3351c3976739ff652159aa1e7928fa0)

**Comment by facebook-github-bot-2** [comment]
@satya164 updated the pull request - [view changes](https://github.com/facebook/react-native/pull/9891/files/329509ebd3351c3976739ff652159aa1e7928fa0..e8142fe8aee5acd0e3ff78ef95681ede89e456f0)

**Comment by satya164** [comment]
Ping @bestander :)

File: Libraries/Components/Intent/IntentAndroid.android.js [hunk 1]
```diff
@@ -1,142 +0,0 @@
...
- *
- *    <!-- Accepts URIs that begin with "http://www.facebook.com/react -->
- *    <data android:scheme="http"
- *       android:host="www.facebook.com"
- *       android:pathPrefix="/react" />
- *    <!-- note that the leading "/" is required for pathPrefix-->
- *
- *    <!-- Accepts URIs that begin with "facebook://react -->
- *    <!-- <data android:scheme="facebook" android:host="react" /> -->
- *  </intent-filter>
```
Comment (javache) [inline Libraries/Components/Intent/IntentAndroid.android.js]: This documentation needs to go somewhere.

**Comment by davidaurelio** [comment]
Sad news is that many of these modules are still used internally.

**Comment by satya164** [comment]
@davidaurelio How do you guys use them with all the warnings! 😱

File: Libraries/Components/Intent/IntentAndroid.android.js (hunk 1 above)
Comment (satya164) [inline Libraries/Components/Intent/IntentAndroid.android.js]: @javache we've linked to the official docs here - https://facebook.github.io/react-native/docs/linking.html which should be fine I think

**Comment by hramos** [comment]
Thanks for the cleanup. Any objections to merging this in @bestander?

**Comment by bestander** [comment]
@javache is updating a bunch of internal code for this, it is in progress

**Pull request closed without merging**